Vungle/VungleRouter.m

```diff
@@ -588,44 +624,78 @@ - (VungleAdSize)getVungleBannerAdSizeType:(CGSize)size
+                                                                               eventID:eventID];
+    if (!targetDelegate) {
+        targetDelegate = [self getBannerDelegateWithPlacement:placementID eventID:eventID withBannerState:state];
+    }
+    
+    return targetDelegate;
+}
+
+- (id<VungleRouterDelegate>)getFullScreenDelegateWithPlacement:(NSString *)placementID
+                                                       eventID:(NSString *)eventID
+{
     if (!placementID.length) {
         return nil;
     }
+    
+    id<VungleRouterDelegate> targetDelegate;
+    if (eventID.length > 0) {
+        targetDelegate = [self.hbDelegatesDict objectForKey:eventID];
+    } else {
+        targetDelegate = [self.delegatesDict objectForKey:placementID];
+    }
+    
+    return targetDelegate;
+}
 
-    @synchronized (self) {
-        id<VungleRouterDelegate> targetDelegate = [self.delegatesDict objectForKey:placementID];
-        if (!targetDelegate) {
-            id<VungleRouterDelegate> bannerDelegate =
-            [self getBannerDelegateWithPlacement:placementID
-                                 withBannerState:state];
-            if (bannerDelegate) {
-                targetDelegate = bannerDelegate;
-            }
-        }
-        return targetDelegate;
+- (id<VungleRouterDelegate>)getBannerDelegateWithPlacement:(NSString *)placementID
+                                                       eventID:(NSString *)eventID
+{
+    if (!placementID.length) {
+        return nil;
     }
+    
+    id<VungleRouterDelegate> targetDelegate;
+    if (eventID.length > 0) {
+        targetDelegate = [self.hbBannerDelegates objectForKey:eventID];
+    } else {
+        targetDelegate = [self.bannerDelegates objectForKey:placementID];
+    }
+    
+    return targetDelegate;
 }
 
 - (id<VungleRouterDelegate>)getBannerDelegateWithPlacement:(NSString *)placementID
+                                                   eventID:(NSString *)eventID
                                            withBannerState:(BannerRouterDelegateState)state
 {
-    id<VungleRouterDelegate> bannerDelegate =
-                [self.bannerDelegates objectForKey:placementID];
-    if (bannerDelegate && bannerDelegate.bannerState == state) {
-        return bannerDelegate;
+    id<VungleRouterDelegate> targetDelegate = [self getBannerDelegateWithPlacement:placementID eventID:eventID];
+    if (targetDelegate.bannerState != state) {
+        return nil;
     }
-    return nil;
+
+    return targetDelegate;
 }
 
 - (void)replaceOldBannerDelegateWithDelegate:(id<VungleRouterDelegate>)delegate
-                             withPlacementID:(NSString *)placementID
 {
     @synchronized (self) {
-        id<VungleRouterDelegate> bannerDelegate =
-                    [self.bannerDelegates objectForKey:placementID];
+        NSString *key = [delegate getEventId];
+        NSMapTable<NSString *, id<VungleRouterDelegate>> *delegateTable;
+        if (key) {
```

Comment:
Might we need to check key.length here? Thanks :)

Reply:
Updated.